### PR TITLE
Add typespecs

### PR DESCRIPTION
**Pull request by mohammedzeglam-pg**
related to #35 

I added much more stuffs sorry.

### CODE REVIEW

File: lib/vox/builder/file_writer.ex [hunk 1]
```diff
@@ -6,6 +6,7 @@ defmodule Vox.Builder.FileWriter do
 
   alias Vox.Builder
 
+  @spec write(Vox.Builder.t()) :: :ok
```
Comment (geolessel) [inline lib/vox/builder/file_writer.ex:9]:
```suggestion
  @spec write(Builder.t()) :: :ok
```

File: lib/vox/builder/file.ex [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+  @type t :: %__MODULE__{
+          bindings: keyword(),
+          collections: list(),
+          content: binary(),
+          destination_path: binary(),
+          final: binary(),
+          root_template: binary(),
+          source_path: binary(),
+          template: binary(),
+          compiled: any(),
```
Comment (mohammedzeglam-pg) [inline lib/vox/builder/file.ex:12]: what about compiled type

Comment (geolessel) [reply lib/vox/builder/file.ex:12]: `compiled` stores the result of running `EEx.compile_file/2` which [according to the source](https://github.com/elixir-lang/elixir/blob/9fd97c466301baa74b309f3ea62215a12911c4d8/lib/eex/lib/eex.ex#L255), returns a `Macro.t()`. So
```suggestion
          compiled: Macro.t(),
```

**Review by geolessel — APPROVED** [review]
Thank you for your addition. You are officially the first contributor. ❤️

**Comment by mohammedzeglam-pg** [comment]
> Thank you for your addition. You are officially the first contributor. ❤️

Your welcome, and  I hope not be last one i plain to build my blog with it.

**Pull request merged**